src/js/components/ServiceForm.js

```diff
@@ -33,16 +38,39 @@ class ServiceForm extends SchemaForm {
     Hooks.doAction('serviceFormMount', this);
   }
 
-  handleFormChange() {
+  handleFormChange(model, eventObj) {
+    // Since the keys we're watching are of the
+    // form `prop[id].key`, extract the key
+    let propKey = FormUtil.getPropKey(eventObj.fieldName);
+
+    let shouldUpdateDefinition = Object.keys(model).some(function (changeKey) {
```

Comment:
before we do all of this, can we make sure that it's a change event? `handleFormChange` will handle all events atm like blur, focus, etc. we can do a check to make sure it's a change event to save us some performance